examples/google-cloud-pubsub/lib/google/cloud/pubsub/v1/publisher_client.rb

```diff
@@ -0,0 +1,713 @@
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# EDITING INSTRUCTIONS
+# This file was generated from the file
+# https://github.com/googleapis/googleapis/blob/master/google/pubsub/v1/pubsub.proto,
+# and updates to that file get reflected here through a refresh process.
+# For the short term, the refresh process will only be runnable by Google
+# engineers.
+
+require "json"
+require "pathname"
+
+require "google/gax"
+
+require "google/iam/v1/iam_policy_pb"
+require "google/pubsub/v1/pubsub_pb"
+require "google/cloud/pubsub/v1/credentials"
+
+module Google
+  module Cloud
+    module Pubsub
+      module V1
+        # The service that an application uses to manipulate topics, and to send
+        # messages to a topic.
+        #
+        # @!attribute [r] iam_policy_stub
+        #   @return [Google::Iam::V1::IAMPolicy::Stub]
+        # @!attribute [r] publisher_stub
+        #   @return [Google::Pubsub::V1::Publisher::Stub]
+        class PublisherClient
+          attr_reader :iam_policy_stub, :publisher_stub
+
+          # The default address of the service.
+          SERVICE_ADDRESS = "pubsub.googleapis.com".freeze
+
+          # The default port of the service.
+          DEFAULT_SERVICE_PORT = 443
+
+          # The default set of gRPC interceptors.
+          GRPC_INTERCEPTORS = []
+
+          DEFAULT_TIMEOUT = 30
+
+          PAGE_DESCRIPTORS = {
+            "list_topics" => Google::Gax::PageDescriptor.new(
+              "page_token",
+              "next_page_token",
+              "topics"),
+            "list_topic_subscriptions" => Google::Gax::PageDescriptor.new(
+              "page_token",
+              "next_page_token",
+              "subscriptions")
+          }.freeze
+
+          private_constant :PAGE_DESCRIPTORS
+
+          BUNDLE_DESCRIPTORS = {
+            "publish" => Google::Gax::BundleDescriptor.new(
+              "messages",
+              [
+                "topic"
+              ],
+              subresponse_field: "message_ids")
+          }.freeze
+
+          private_constant :BUNDLE_DESCRIPTORS
+
+          # The scopes needed to make gRPC calls to all of the methods defined in
+          # this service.
+          ALL_SCOPES = [
+            "https://www.googleapis.com/auth/cloud-platform",
+            "https://www.googleapis.com/auth/pubsub"
+          ].freeze
+
+          # Returns a fully-qualified topic resource name string.
+          # @param project [String]
+          # @param topic [String]
+          # @return [String]
+          def self.topic_path project, topic
```

Comment:
Are we going to add instance methods for all of these? I suppose we have to keep the class level ones as well.